docs/develop/library.rst

```diff
@@ -0,0 +1,234 @@
+.. _internals_library:
+
+Implementing a MicroPython Module
+==================================
+
+This chapter details how to implement a C module that is not a builtin module. It also
+explains how to implement and distribute a MicroPython package.
+
+A non-builtin C module
+----------------------
+
+A C module in MicroPython can be a builtin and part of the core builtins or be outside the core library. See :ref:`internals_builtin_modules` for details on how to 
```

Comment:
I think the terms "core library" and "standard library" are perhaps a bit confusing here (and in the next para)... Also "builtins" is an overloaded term (mixed up with Python's "builtins" module).

Perhaps instead explain just what the three options are:
 - built-in
 - user
 - dynamic

And explain an example of where each would be used:
 - built-in -- a general module that should be part of the MicroPython repository
 - user -- a module that is useful for your specific project (i.e. project-specific functionality that you maintian in your own repository or private codebase)
 - dynamic -- a module that can be deployed at runtime to your device

We don't need to provide a recommendation, rather the reader can figure out what applies to them.